openmc/filter.py

```diff
@@ -0,0 +1,1810 @@
+            cell instance ID for 'distribcell' Filters. The bin is a 2-tuple of
+            floats for 'energy' and 'energyout' filters corresponding to the
+            energy boundaries of the bin of interest. The bin is an (x,y,z)
+            3-tuple for 'mesh' filters corresponding to the mesh cell of
+            interest.
+
+        Returns
+        -------
+        filter_index : int
+             The index in the Tally data array for this filter bin.
+
+        """
+
+        if filter_bin not in self.bins:
+            msg = 'Unable to get the bin index for Filter since "{0}" ' \
+                  'is not one of the bins'.format(filter_bin)
+            raise ValueError(msg)
+
+        if isinstance(self.bins, np.ndarray):
+            return np.where(self.bins == filter_bin)[0][0]
+        else:
+            return self.bins.index(filter_bin)
+
+    def get_pandas_dataframe(self, data_size, stride, **kwargs):
+        """Builds a Pandas DataFrame for the Filter's bins.
+
+        This method constructs a Pandas DataFrame object for the filter with
+        columns annotated by filter bin information. This is a helper method for
+        :meth:`Tally.get_pandas_dataframe`.
+
+        Parameters
+        ----------
+        data_size : int
+            The total number of bins in the tally corresponding to this filter
+        stride : int
+            Stride in memory for the filter
+
+        Keyword arguments
+        -----------------
+        paths : bool
+            Only used for DistribcellFilter.  If True (default), expand
+            distribcell indices into multi-index columns describing the path
+            to that distribcell through the CSG tree.  NOTE: This option assumes
+            that all distribcell paths are of the same length and do not have
+            the same universes and cells but different lattice cell indices.
+
+        Returns
+        -------
+        pandas.DataFrame
+            A Pandas DataFrame with columns of strings that characterize the
+            filter's bins. The number of rows in the DataFrame is the same as
+            the total number of bins in the corresponding tally, with the filter
+            bin appropriately tiled to map to the corresponding tally bins.
+
+        See also
+        --------
+        Tally.get_pandas_dataframe(), CrossFilter.get_pandas_dataframe()
+
+        """
+        # Initialize Pandas DataFrame
+        df = pd.DataFrame()
+
+        filter_bins = np.repeat(self.bins, stride)
+        tile_factor = data_size // len(filter_bins)
+        filter_bins = np.tile(filter_bins, tile_factor)
+        df = pd.concat([df, pd.DataFrame(
+            {self.short_name.lower(): filter_bins})])
+
+        return df
+
+
+class WithIDFilter(Filter):
+    """Abstract parent for filters of types with IDs (Cell, Material, etc.)."""
+    def __init__(self, bins, filter_id=None):
+        bins = np.atleast_1d(bins)
```

Comment:
This `__init__` does not make a call to the superclass' `__init__`. Is this intention and/or desired? I would recommend calling the superclass' init function and remove the last two lines of this one (which is handled by the superclass)